app/controlplane/pkg/biz/user.go

```diff
@@ -90,6 +96,15 @@ func (uc *UserUseCase) DeleteUser(ctx context.Context, userID string) error {
 	return uc.userRepo.Delete(ctx, userUUID)
 }
 
+func (uc *UserUseCase) CreateByEmail(ctx context.Context, email string) (*User, error) {
```

Comment:
I have mixed feelings of creating methods just for testing (same about findbyname), it feels upside down to me. 

In fact I think in this case this is wrong and can be confusing, it's not trivial to know that `createbyemail` will not add auto-onboarding. 

Personally, I've have passed an extra optional argument `WithDisableAutoOnboarding` or make it part of the the userUsecase struct`

Reply:
I would wanted it to be more explicit on the auto but adding it to the FindOrCreate does not harm much as long as we always remember that the autonboarding exists there.

I've added an optional parameter to the method to disable the auto onboarding. Please note that I didn't create a regular Builder with all Its options because I do believe it's too verbose for just one configuration.